sechub-storage-sharedvolume-spring/src/main/java/com/daimler/sechub/storage/sharevolume/spring/SharedVolumeJobStorage.java

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.storage.sharevolume.spring;
+
+import static java.util.Objects.*;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.FileSystemUtils;
+import org.springframework.web.client.ResourceAccessException;
+
+import com.daimler.sechub.storage.core.JobStorage;
+import com.daimler.sechub.storage.core.StorageException;
+
+public class SharedVolumeJobStorage implements JobStorage {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SharedVolumeJobStorage.class);
+    private String storagePath;
+    private UUID jobUUID;
+    private Path volumePath;
+
+    public SharedVolumeJobStorage(Path rootLocation, String storagePath, UUID jobUUID) {
+        requireNonNull(rootLocation, "rootlocation may not be null");
+        requireNonNull(storagePath, "projectId may not be null");
+        requireNonNull(jobUUID, "jobUUID may not be null");
+
+        this.storagePath = storagePath;
+        this.jobUUID = jobUUID;
+
+        this.volumePath = rootLocation.resolve(storagePath).resolve(jobUUID.toString());
+    }
+
+    @Override
+    public InputStream fetch(String name) throws IOException {
+        Path path = getPathToFile(name);
+        if (path == null) {
+            return null;
+        }
+        return new FileInputStream(path.toFile());
+    }
+
+    @Override
+    public void store(String name, InputStream stream) throws IOException {
+        requireNonNull(name, "name may not be null!");
+        requireNonNull(stream, "stream may not be null!");
+
+        if (name.contains("..")) {
+            // This is a security check
+            throw new StorageException("Cannot store file with relative path outside current directory " + name);
+        }
+
+        try {
+            Files.createDirectories(volumePath);
+        } catch (IOException e) {
+            throw new StorageException("Could not initialize storage directory at:" + volumePath, e);
+        }
+        LOG.info("job:{}: storing {} in path {}", jobUUID, name, storagePath);
+
+        try (InputStream inputStream = stream) {
+            Path pathToFile = getPathToFile(name);
+            Files.copy(inputStream, pathToFile, StandardCopyOption.REPLACE_EXISTING);
+            LOG.debug("Stored:{} at {}", name, pathToFile);
+        } catch (ResourceAccessException e) {
+            throw new IOException("Provided file exceeds file limit.", e);
```

Comment:
```suggestion
            throw new IOException("Provided file exceeds file size limit.", e);
```

Reply:
After looking at the code, It's not about file size here (that could be ONE reason), but the resource access exception will contain the information whats happening.

So changed to
```
throw new IOException("Was not able to store input stream into file "+ pathToFile, e);
```
Which explains more what is happening here. Details must be inside exception cause (e)